app-drawer/app-drawer.html

```diff
@@ -120,37 +116,156 @@
+
+        _translateOffset: Number
 
       },
 
+      listeners: {
+        tap: '_tapHandler',
+        track: '_trackHandler'
+      },
+
       /**
-       * Toggles the drawer in and out.
-       *
-       * @method toggle
+       * Opens the drawer.
+       */
+      open: function() {
+        this.opened = true;
+      },
+
+      /**
+       * Closes the drawer.
+       */
+      close: function() {
+        this.opened = false;
+      },
+
+      /**
+       * Toggles the drawer open and close.
        */
       toggle: function() {
         this.opened = !this.opened;
+      },
+
+      /**
+       * Gets the width of the drawer.
+       *
+       * @return {number} The width of the drawer in pixels.
+       */
+      getDrawerWidth: function() {
+        return this.$.drawerContent.offsetWidth;
+      },
+
+      _tapHandler: function() {
+        if (!this.persistent) {
+          this.opened = false;
+        }
+      },
+
+      _trackHandler: function(event) {
+        if (!this.persistent) {
+          switch (event.detail.state) {
+            case 'start':
+              this.$.drawerContent.style.transitionDuration = '0s';
+              this.$.drawerContent.style.webkitTransitionDuration = '0s';
+
+              if (this.opened) {
+                this._translateOffset = 0;
+              } else if (this.position == 'left') {
+                this._translateOffset = -this.getDrawerWidth();
+              } else if (this.position == 'right') {
+                this._translateOffset = this.getDrawerWidth();
+              }
+              break;
+
+            case 'track':
+              switch (this.position) {
+                case 'left':
+                  this._translateDrawer(Math.min(0, event.detail.dx + this._translateOffset));
+                  break;
+                case 'right':
+                  this._translateDrawer(Math.max(0, event.detail.dx + this._translateOffset));
+                  break;
+              }
+              if (Math.abs(event.detail.ddx) > 0) {
+                this._lastTrackDirection = event.detail.ddx;
+              }
+              break;
+
+            case 'end':
+              this.$.drawerContent.style.transitionDuration = '';
+              this.$.drawerContent.style.webkitTransitionDuration = '';
+              this.transform('', this.$.drawerContent);
+              this.opened = (this.position === 'left' && this._lastTrackDirection > 0) ||
+                (this.position === 'right' && this._lastTrackDirection < 0);
+              break;
+          }
+        }
+      },
+
+      _openedChanged: function() {
+        this._updateDocScroll();
+      },
+
+      _persistentChanged: function() {
+        this._updateDocScroll();
+      },
+
+      _translateDrawer: function(x) {
+        this.transform('translateX(' + x + 'px)', this.$.drawerContent);
+      },
+
+      _updateDocScroll: function() {
+        this.shouldEnableDocScroll(this.persistent || !this.opened);
```

Comment:
`overflow: hidden` on supported browsers (not mobile Safari); remove `AppDocumentScrollBehavior`